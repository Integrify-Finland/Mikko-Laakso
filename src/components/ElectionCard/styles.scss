.blog-card {
  color: $primary;
  display: flex;
  flex-direction: column;
  margin: 1rem auto;
  box-shadow: 0 3px 7px -1px rgba($dark, 0.1);
  margin-bottom: 1.6%;
  background: $white;
  line-height: 1.4;
  font-family: $font-family;
  border-radius: 5px;
  overflow: hidden;
  z-index: 0;

  a {
    color: inherit;
    &:hover {
      color: $primary;
    }
  }

  &:hover &__photo {
    transform: scale(1.3) rotate(3deg);
  }
  &__meta {
    position: relative;
    z-index: 0;
    height: 200px;
  }
  &__photo {
    background-size: cover;
    background-position: center;
    transition: transform 0.2s;
    height: 100%;
  }
  &__details,
  &__details ul {
    margin: auto;
    padding: 0;
    list-style: none;
  }

  &__details {
    position: absolute;
    top: 0;
    bottom: 0;
    left: -100%;
    margin: auto;
    transition: left 0.2s;
    background: rgba($dark, 0.6);
    color: $white;
    padding: $spacing-00-75;
    width: 100%;
    font-size: 0.9rem;
    a {
      text-decoration: dotted underline;
    }
    ul li {
      display: inline-block;
    }
    .author:before {
      font-family: $font-family;
      margin-right: 10px;
      content: '\f007';
    }

    .tags {
      ul:before {
        font-family: $font-family;
        content: '\f02b';
        margin-right: 10px;
      }
      li {
        margin-right: 2px;
        &:first-child {
          margin-left: -4px;
        }
      }
    }
  }
  &__description {
    padding: 1rem;
    background: $white;
    position: relative;
    z-index: 1;
    h1,
    h2 {
      font-family: $font-family;
    }
    h1 {
      line-height: 1;
      margin: 0;
      font-size: 1.7rem;
    }
    h2 {
      font-size: 1rem;
      font-weight: 300;
      text-transform: uppercase;
      color: $grey;
      margin-top: 5px;
    }
  }
  &__description &__read-more {
    text-align: right;
    .link {
      color: $primary;
      position: relative;
      &:after {
        content: '';

        font-family: $font-family;
        margin-left: -10px;
        opacity: 0;
        vertical-align: middle;
        transition: margin 0.3s, opacity 0.3s;
      }

      &:hover:after {
        margin-left: 5px;
        opacity: 1;
      }
    }
  }
  p {
    position: relative;
    margin: 1rem 0 0;
    &:first-of-type {
      margin-top: 1.25rem;
      &:before {
        content: '';
        position: absolute;
        height: 5px;
        background: $primary;
        width: 35px;
        top: -0.75rem;
        border-radius: 3px;
      }
    }
  }
  &:hover &__details {
    left: 0%;
  }

  @media (min-width: 900px) {
    flex-direction: row;
    max-width: 900px;
    &__meta {
      flex-basis: 30%;
      height: auto;
    }
    &__description {
      flex-basis: 70%;
      &:before {
        transform: skewX(-3deg);
        content: '';
        background: $white;
        width: 30px;
        position: absolute;
        left: -10px;
        top: 0;
        bottom: 0;
        z-index: -1;
      }
    }
    &.alt & {
      flex-direction: row-reverse;
      &__description {
        &:before {
          left: inherit;
          right: -10px;
          transform: skew(3deg);
        }
      }
      &__details {
        padding-left: $spacing-01-50;
      }
    }
  }
  @include for-phone {
    width: 90%;
  }
}
