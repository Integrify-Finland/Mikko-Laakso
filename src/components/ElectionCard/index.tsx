import React, { FC } from 'react'
import { Link } from 'gatsby'
import './styles.scss'
interface Props {
	/** Background image of card */
	backGroundImage: string
	/** Title of eclection theme article */
	title: string
	/** Highlighted statement in article */
	emphasisStatement: string
	/** Article content */
	description: string
	/** Published date */
	date: string
	/** Name of author */
	author: string
	/** Link to Election theme page to view full content */
	url: string
}
const ElectionCard: FC<Props> = ({
	backGroundImage,
	title,
	emphasisStatement,
	description,
	date,
	author,
	url,
}) => {
	return (
		<div className="blog-card">
			<div className="meta">
				<div
					className="photo"
					style={{
						backgroundImage: `url(${backGroundImage})`,
					}}
				/>

				<ul className="details">
					<li className="author">{author}</li>
					<li className="date">{date}</li>
				</ul>
			</div>
			<div className="description">
				<h1>{title}</h1>
				<h2>{emphasisStatement}</h2>
				<p> {description}</p>

				<p className="read-more">
					<Link className="link" to={url}>
						Read More
					</Link>
				</p>
			</div>
		</div>
	)
}
ElectionCard.defaultProps = {
	backGroundImage:
		'https://mikkolaakso.com/wp-content/uploads/2015/08/Espoon_Tapiola_kes%C3%A4ll%C3%A4.jpg',
	title: 'Itsenäisen Espoon puolesta',
	emphasisStatement: 'Itsenäinen Espoo on metropolialueen kehityksen etu.',
	description:
		'Espoolaisten kuntapäättäjien tulee antaa voimakas viesti sen puolesta, ettei ole kenenkään edun mukaista rakentaa pääkaupunkiseudulle uutta tehotonta mammuttikuntaa.',
	date: '5th Nov 2019',
	author: 'Mikko Laakso',
	url: '/',
}
export default ElectionCard
