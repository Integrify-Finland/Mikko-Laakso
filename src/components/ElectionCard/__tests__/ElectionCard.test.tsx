import React from 'react'
import ElectionCard from '..'
import { render } from '@testing-library/react'

describe('<ElectionCard />', () => {
	it('should render', () => {
		const { container } = render(
			<ElectionCard
				backGroundImage={
					'https://mikkolaakso.com/wp-content/uploads/2015/08/Espoon_Tapiola_kes%C3%A4ll%C3%A4.jpg'
				}
				title={'Itsenäisen Espoon puolesta'}
				emphasisStatement={
					'Itsenäinen Espoo on metropolialueen kehityksen etu.'
				}
				description={
					'Espoolaisten kuntapäättäjien tulee antaa voimakas viesti sen puolesta, ettei ole kenenkään edun mukaista rakentaa pääkaupunkiseudulle uutta tehotonta mammuttikuntaa.'
				}
				date={'5th Nov 2019'}
				author={'Mikko Laakso'}
				url={'/'}
			/>,
		)

		expect(container).toMatchSnapshot()
	})
})
