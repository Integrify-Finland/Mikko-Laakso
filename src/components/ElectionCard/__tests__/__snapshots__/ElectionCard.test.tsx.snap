// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ElectionCard /> should render 1`] = `
<div>
  <div
    class="blog-card"
  >
    <div
      class="meta"
    >
      <div
        class="photo"
        style="background-image: url(https://mikkolaakso.com/wp-content/uploads/2015/08/Espoon_Tapiola_kes%C3%A4ll%C3%A4.jpg);"
      />
      <ul
        class="details"
      >
        <li
          class="author"
        >
          Mikko Laakso
        </li>
        <li
          class="date"
        >
          5th Nov 2019
        </li>
      </ul>
    </div>
    <div
      class="description"
    >
      <h1>
        Itsenäisen Espoon puolesta
      </h1>
      <h2>
        Itsenäinen Espoo on metropolialueen kehityksen etu.
      </h2>
      <p>
         
        Espoolaisten kuntapäättäjien tulee antaa voimakas viesti sen puolesta, ettei ole kenenkään edun mukaista rakentaa pääkaupunkiseudulle uutta tehotonta mammuttikuntaa.
      </p>
      <p
        class="read-more"
      >
        <a
          class="link"
          href="/"
        >
          Read More
        </a>
      </p>
    </div>
  </div>
</div>
`;
